.ci/jenkins/Jenkinsfile.init-branch

```diff
@@ -0,0 +1,154 @@
+import org.jenkinsci.plugins.workflow.libs.Library
+@Library('jenkins-pipeline-shared-libraries')_
+
+import org.kie.jenkins.MavenCommand
+
+pipeline {
+    agent {
+        label 'kie-rhel7 && kie-mem16g && !master'
+    }
+
+    tools {
+        maven env.BUILD_MAVEN_TOOL
+        jdk env.BUILD_JDK_TOOL
+    }
+
+    options {
+        timestamps()
+        timeout(time: 60, unit: 'MINUTES')
+    }
+
+    // parameters {
+    // For parameters, check into ./dsl/jobs.groovy file
+    // }
+
+    environment {
+        // Static env is defined into ./dsl/jobs.groovy file
+
+        OPTAPLANNER_CI_EMAIL_TO = credentials("${JENKINS_EMAIL_CREDS_ID}")
+
+        // Keep here for visitibility
+        MAVEN_OPTS = '-Xms1024m -Xmx4g'
+    }
+
+    stages {
+        stage('Initialize') {
+            steps {
+                script {
+                    cleanWs()
+
+                    if (params.DISPLAY_NAME) {
+                        currentBuild.displayName = params.DISPLAY_NAME
+                    }
+
+                    checkoutRepo(getRepoName(), getBuildBranch())
+                }
+            }
+        }
+        stage('Update project version') {
+            steps {
+                script {
+                    maven.mvnVersionsSet(getMavenCommand(getRepoName()), getOptaPlannerVersion(), true)
+                }
+            }
+        }
+        stage('Deploy to repository') {
```

Comment:
What is the role of init-branch jobs in the release process? I am a bit surprised to see the deploy stage here, which is a part of the `Jenkinsfile.deploy`.

Reply:
The init branch is called once the release branch has just been created. So we can setup the correct version.
Once that is done, we need to deploy the artifacts to Nexus so that anybody checking out downstream repositories on that new branch could use them directly.